qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -210,7 +216,18 @@ def run(self, dag):
             self.property_set["layout"] = initial_layout
             self.routing_pass.fake_run = False
             return dag
+
         dist_matrix = self.coupling_map.distance_matrix
+        print('original distance matrix')
+        print(dist_matrix)
+
+        if self.target:
+            dist_matrix = rx.digraph_floyd_warshall_numpy(
```

Comment:
Do you think that it makes sense to wrap this in a try/except if the target basis isn't known to `TwoQubitWeylDecomposition` and fall back to the default distance matrix? I'm just wondering what happens if the decomposer doesn't handle the 2q gates in the basis?

Reply:
yeah you are right i still need to write test cases for the different cases we may encounter